python/simulation.py

```diff
@@ -4996,14 +5004,16 @@ def _combine(sim, todo):
 
 def _eval_step_func(sim, func, todo):
     num_args = get_num_args(func)
+    name_args = get_name_args(func)
+    self_count = int("self" in name_args)
```

Comment:
Using `inspect.ismethod(func)` seems like a more reliable way to check for this.  However, this seems unreliable in another way: it assumes that any bound method is a method of the `sim` object.    What if it is a method of some other object?